From ecb3a4b567dbc4ea1ccebea629ad5f09ff77b067 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 31 Oct 2025 21:26:04 +0100
Subject: [PATCH] Fix colors in HighContrast theme

This fixes the following problems with the built-in HighContrast theme:
- White text over white background in the main menu.
- Black text over black background on the selecred item.

Also, reset the sytle properties similar to the normal theme, except the
borders, which is important for a high contrast theme. This fixes some
low contrast items, e.g. selected category in the main menu.
---
 src/theme/3.20/theme_hc.scss  | 17 +++++++++++++++++
 src/theme/common/_colors.scss |  2 +-
 src/theme/common/_menu.scss   |  4 ++--
 3 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/src/theme/3.20/theme_hc.scss b/src/theme/3.20/theme_hc.scss
index 2ec8d8a5c..f5f5f2d35 100644
--- a/src/theme/3.20/theme_hc.scss
+++ b/src/theme/3.20/theme_hc.scss
@@ -1,5 +1,22 @@
 $variant: 'hc';
 
+// Used to reset all style properties set by the gtk theme
+%reset_style {
+    background: none;
+    box-shadow: none;
+    margin: 0;
+    text-shadow: none;
+    -gtk-icon-shadow: none;
+    -gtk-icon-transform: none;
+    opacity: 1;
+    min-width: 0;
+    min-height: 0;
+    padding: unset;
+    font-size: unset;
+    font-weight: unset;
+}
+
+
 @import 'settings';
 @import '../common/imports';
 @import 'widgets';
diff --git a/src/theme/common/_colors.scss b/src/theme/common/_colors.scss
index 5a1abcf18..ac98b8c07 100644
--- a/src/theme/common/_colors.scss
+++ b/src/theme/common/_colors.scss
@@ -1,7 +1,7 @@
 $fg_color: if($variant == 'default', white, black);
 
 $selected_bg_color: if($variant =='default', rgb(107, 202, 129), black);
-$selected_fg_color: black;
+$selected_fg_color: if($variant == "default", black, white);
 
 $warning_color: #F27835;
 $error_color: #FC4138;
diff --git a/src/theme/common/_menu.scss b/src/theme/common/_menu.scss
index 0411649e0..cee8f6e89 100644
--- a/src/theme/common/_menu.scss
+++ b/src/theme/common/_menu.scss
@@ -7,10 +7,10 @@
     .button:hover { -gtk-#{$image-effect}: none; }
 
     label, #{$search} {
-        color: darken(white, 0.1);
+        color: darken($fg_color, 0.1);
 
         &:disabled {
-            color: transparentize(white, 0.4);
+            color: transparentize($fg_color, 0.4);
         }
     }
 
